fs/mmap/fs_mmap.c

```diff
@@ -272,11 +252,11 @@ int file_mmap(FAR struct file *filep, FAR void *start, size_t length,
 FAR void *mmap(FAR void *start, size_t length, int prot, int flags,
                int fd, off_t offset)
 {
-  FAR struct file *filep;
+  FAR struct file *filep = NULL;
   FAR void *mapped;
   int ret;
 
-  if (fs_getfilep(fd, &filep) < 0)
+  if (fd != -1 && fs_getfilep(fd, &filep) < 0)
```

Comment:
fd != -1 is not needed, since fs_getfilep returns -EBADF in canse fd is invalid

Reply:
But, we need support instead return -EBADF:
ptr = mmap(NULL, length, PROT_WRITEt, MAP_ANONYMOUS, -1, 0);

Reply:
of course, my bad